cpp/src/arrow/compute/kernels/vector_sort.cc

```diff
@@ -2340,6 +2340,9 @@ class SelectKUnstableMetaFunction : public MetaFunction {
       return Status::Invalid("SelectK requires a nonnegative `k`, got ",
                              select_k_options.k);
     }
+    if (select_k_options.sort_keys.size() == 0) {
+      return Status::Invalid("SelectK requires `sort_keys` option paremeter, got {}");
```

Comment:
I would maybe leave out the `got {}`, as this will look a bit strange for users of the bindings (the {} stands for an empty vector in C++, but Python/R users won't necessarily understand that)

Reply:
Right. Instead, make the error message explicit: "SelectK requires a non-empty `sort_keys`".